src/store/adjustVault.ts

```diff
@@ -0,0 +1,231 @@
+import { calculateCurrentDebt } from '@agoric/inter-protocol/src/interest-math';
+import {
+  addRatios,
+  ceilMultiplyBy,
+  makeRatioFromAmounts,
+} from '@agoric/zoe/src/contractSupport';
+import { ratioGTE } from '@agoric/zoe/src/contractSupport/ratio';
+import { atom } from 'jotai';
+import { pursesAtom } from './app';
+import { vaultKeyToAdjustAtom, vaultStoreAtom } from './vaults';
+
+export const vaultToAdjustAtom = atom(get => {
+  const key = get(vaultKeyToAdjustAtom);
+  const { vaults, vaultManagers, prices, vaultGovernedParams, vaultMetrics } =
+    get(vaultStoreAtom);
+
+  const vault = key && vaults?.get(key);
+  if (!vault) {
+    return null;
+  }
+
+  const { locked, debtSnapshot, managerId, createdByOfferId } = vault;
+  const manager = vaultManagers.get(managerId);
+  const price = locked && prices.get(locked.brand);
+  const params = vaultGovernedParams.get(managerId);
+  const metrics = vaultMetrics.get(managerId);
+  if (!(locked && debtSnapshot && manager && price && params && metrics)) {
+    return null;
+  }
+
+  const totalLockedValue = ceilMultiplyBy(
+    locked,
+    makeRatioFromAmounts(price.amountOut, price.amountIn),
+  );
+
+  const totalDebt = calculateCurrentDebt(
+    debtSnapshot.debt,
+    debtSnapshot.interest,
+    manager.compoundedInterest,
+  );
+
+  return {
+    totalLockedValue,
+    totalDebt,
+    collateralPrice: price,
+    locked,
+    indexWithinManager: vault.indexWithinManager,
+    params,
+    metrics,
+    collateralizationRatio: makeRatioFromAmounts(totalLockedValue, totalDebt),
+    createdByOfferId,
+  };
+});
+
+export const collateralDeltaValueAtom = atom<bigint | null>(null);
+
+export const debtDeltaValueAtom = atom<bigint | null>(null);
+
+export enum CollateralAction {
+  None = 'No Action',
+  Deposit = 'Deposit',
+  Withdraw = 'Withdraw',
+}
+
+export enum DebtAction {
+  None = 'No Action',
+  Repay = 'Repay',
+  Borrow = 'Borrow More',
+}
+
+const collateralActionAtomInternal = atom(CollateralAction.None);
+
+export const collateralActionAtom = atom(
+  get => get(collateralActionAtomInternal),
+  (_get, set, action: CollateralAction) => {
+    set(collateralDeltaValueAtom, null);
+    set(collateralActionAtomInternal, action);
+  },
+);
+
+const debtActionAtomInternal = atom(DebtAction.None);
+
+export const debtActionAtom = atom(
+  get => get(debtActionAtomInternal),
+  (_get, set, action: DebtAction) => {
+    set(debtDeltaValueAtom, null);
+    set(debtActionAtomInternal, action);
+  },
+);
+
+export const vaultAfterAdjustmentAtom = atom(get => {
+  const vaultToAdjust = get(vaultToAdjustAtom);
+  if (!vaultToAdjust) {
+    return null;
+  }
+
+  const debtAction = get(debtActionAtom);
+  const collateralAction = get(collateralActionAtom);
+  const collateralDeltaValue = get(collateralDeltaValueAtom);
+  const debtDeltaValue = get(debtDeltaValueAtom);
+
+  const { totalDebt, locked, params, collateralPrice } = vaultToAdjust;
+
+  const newDebt = (() => {
```

Comment:
please extract `newDebt` and `newLocked` into `vaultMath.ts` and add some tests

Reply:
done